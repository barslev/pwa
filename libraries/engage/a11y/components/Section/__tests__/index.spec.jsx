import React from 'react';
import { mount } from 'enzyme';
import Section from '../index';

const mockTranslate = jest.fn();

jest.mock('../../../../components', () => ({
  I18n: {
    Text: function Translate(props) {
      mockTranslate(props);
      return props.string;
    },
  },
}));

const title = 'my.headline';
const titleParams = { some: 'params' };

describe('<Section />', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('should render with a label', () => {
    const wrapper = mount((
      <Section title={title} titleParams={titleParams}>
        <div id="child-component" />
      </Section>
    ));

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('h2').exists()).toBe(true);
    expect(mockTranslate).toHaveBeenCalledWith({
      string: title,
      params: titleParams,
    });
  });

  it('should render hidden when no children rendered', () => {
    const wrapper = mount(<Section title={title} />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('h2').exists()).toBe(false);
    expect(mockTranslate).toHaveBeenCalledTimes(0);
  });

  describe('MutationObserver', () => {
    let wrapper;

    beforeEach(() => {
      wrapper = mount(<Section title={title} />);
    });

    it('should create an observer instance', () => {
      expect(global.mutationConstructorSpy).toHaveBeenCalledTimes(1);
      expect(global.mutationConstructorSpy).toHaveBeenCalledWith(expect.any(Function));
    });

    it('should observe with the correct initialization', () => {
      expect(global.mutationObserveSpy).toHaveBeenCalledTimes(1);
      expect(global.mutationObserveSpy).toHaveBeenCalledWith(
        wrapper.find('section').instance(),
        {
          childList: true,
        }
      );
    });

    it('should call disconnect on unmount', () => {
      wrapper.unmount();
      expect(global.mutationDisconnectSpy).toHaveBeenCalledTimes(1);
    });
  });
});
