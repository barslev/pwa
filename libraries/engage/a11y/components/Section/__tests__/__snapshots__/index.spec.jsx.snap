// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Section /> should render hidden when no children rendered 1`] = `
<Section
  className={null}
  title="my.headline"
  titleParams={Object {}}
>
  <section />
</Section>
`;

exports[`<Section /> should render with a label 1`] = `
<Section
  className={null}
  title="my.headline"
  titleParams={
    Object {
      "some": "params",
    }
  }
>
  <section
    aria-labelledby="my-headline"
  >
    <h2
      hidden={true}
      id="my-headline"
    >
      <Translate
        params={
          Object {
            "some": "params",
          }
        }
        string="my.headline"
      >
        my.headline
      </Translate>
    </h2>
    <div
      id="child-component"
    />
  </section>
</Section>
`;
